### PR TITLE
Allow to override background, icon and divider colors

### PR DESCRIPTION
**Pull request by RashidianPeyman**
Hello, dear Niklas Baudy (Vacation)  first of all, I apologize for my English and thank u for your useful third- party. I use your Emoji third-party in my app but I needed to change background, divider and icon colors programmatically based on my themes. so I fork the Emoji and changed it on demand. now I can use three methods to change background, divider and icon colors in setUpEmojiPopup so we have 
this Three :
        .setBackgroundColor(Color.parseColor("#BDBDBD"))
        .setDividerColor(Color.parseColor("#42A5F5"))
        .setIconColor(Color.parseColor("#f0f0f0"))

Although if u don't use them. default values will call and there is no problem

With respect
Peyman Rashidian

### CODE REVIEW

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java [hunk 1]
```diff
@@ -83,24 +87,29 @@
     }
   };
 
-  EmojiPopup(@NonNull final View rootView, @NonNull final EmojiEditTextInterface editInterface,
-            @Nullable final RecentEmoji recent, @Nullable final VariantEmoji variant) {
+  public EmojiPopup(@NonNull final View rootView, @NonNull final EmojiEditTextInterface editInterface,
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java]: Why public?

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java [hunk 2]
```diff
@@ -29,7 +31,8 @@
   static final int MIN_KEYBOARD_HEIGHT = 100;
 
   final View rootView;
-  final Activity context;
+  private Activity context;
+  private int backgroundColor, iconColor, dividerColor;
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java]: Please put each on it's own line

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java [hunk 3]
```diff
@@ -277,17 +291,36 @@ private Builder(final View rootView) {
...
+
+      return this;
+    }
+
+    @CheckResult public Builder setIconColor(final int iconColor) {
+      this.iconColor = iconColor;
+      return this;
+    }
+
+    @CheckResult public Builder setDividerColor(final int dividerColor) {
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java]: Can you add `@ColorInt`?

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java (hunk 1 above)
Comment (RashidianPeyman) [reply emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java]: it was my gaffe, i correct it

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java (hunk 2 above)
Comment (RashidianPeyman) [reply emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java]: done

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java (hunk 3 above)
Comment (RashidianPeyman) [reply emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java]: yes I will add them

**Comment by RashidianPeyman** [comment]
all revert formatting changes did and revisions done

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java [hunk 4]
```diff
@@ -3,17 +3,20 @@
...
 import android.support.annotation.CheckResult;
+import android.support.annotation.ColorInt;
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 import android.view.Gravity;
 import android.view.View;
 import android.view.ViewTreeObserver;
 import android.view.inputmethod.InputMethodManager;
 import android.widget.PopupWindow;
+
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java]: revert this one please

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java [hunk 5]
```diff
@@ -50,7 +56,8 @@
   @Nullable OnEmojiPopupDismissListener onEmojiPopupDismissListener;
 
   final ViewTreeObserver.OnGlobalLayoutListener onGlobalLayoutListener = new ViewTreeObserver.OnGlobalLayoutListener() {
-    @Override public void onGlobalLayout() {
+    @Override
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java]: should be on the same line just as it was previosly

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java [hunk 6]
```diff
@@ -83,24 +90,29 @@
...
     this.variantEmoji = variant != null ? variant : new VariantEmojiManager(context);
+    this.backgroundColor = backgroundColor;
+    this.iconColor = iconColor;
+    this.dividerColor = dividerColor;
 
     popupWindow = new PopupWindow(context);
 
     final OnEmojiLongClickListener longClickListener = new OnEmojiLongClickListener() {
-      @Override public void onEmojiLongClick(@NonNull final EmojiImageView view, @NonNull final Emoji emoji) {
+      @Override
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java]: same

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java [hunk 7]
```diff
@@ -83,24 +90,29 @@
     }
   };
 
-  EmojiPopup(@NonNull final View rootView, @NonNull final EmojiEditTextInterface editInterface,
-            @Nullable final RecentEmoji recent, @Nullable final VariantEmoji variant) {
+   EmojiPopup(@NonNull final View rootView, @NonNull final EmojiEditTextInterface editInterface,
+                    @Nullable final RecentEmoji recent, @Nullable final VariantEmoji variant, int backgroundColor, int iconColor, int dividerColor) {
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java]: same

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java [hunk 8]
```diff
@@ -83,24 +90,29 @@
...
-      @Override public void onEmojiLongClick(@NonNull final EmojiImageView view, @NonNull final Emoji emoji) {
+      @Override
+      public void onEmojiLongClick(@NonNull final EmojiImageView view, @NonNull final Emoji emoji) {
         variantPopup.show(view, emoji);
       }
     };
 
     final OnEmojiClickListener clickListener = new OnEmojiClickListener() {
-      @Override public void onEmojiClick(@NonNull final EmojiImageView imageView, @NonNull final Emoji emoji) {
+      @Override
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java]: same

Comment (RashidianPeyman) [reply emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java]: I tried to do them 😉

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java (hunk 7 above)
Comment (RashidianPeyman) [reply emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java]: done

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java (hunk 6 above)
Comment (RashidianPeyman) [reply emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java]: done

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java (hunk 5 above)
Comment (RashidianPeyman) [reply emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java]: done

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java (hunk 4 above)
Comment (RashidianPeyman) [reply emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java]: done

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java [hunk 9]
```diff
@@ -84,12 +89,15 @@
   };
 
   EmojiPopup(@NonNull final View rootView, @NonNull final EmojiEditTextInterface editInterface,
-            @Nullable final RecentEmoji recent, @Nullable final VariantEmoji variant) {
+            @Nullable final RecentEmoji recent, @Nullable final VariantEmoji variant, int backgroundColor, int iconColor, int dividerColor) {
     this.context = Utils.asActivity(rootView.getContext());
     this.rootView = rootView.getRootView();
-    this.editInterface = editInterface;
+    this.editInterface  = editInterface ;
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java]: Revert

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java [hunk 10]
```diff
@@ -149,7 +157,7 @@ public void toggle() {
       if (isKeyboardOpen) {
         // If the keyboard is visible, simply show the emoji popup.
         showAtBottom();
-      } else if (editInterface instanceof View) {
+      } else if (editInterface  instanceof View) {
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java]: Revert

File: emoji/src/main/java/com/vanniktech/emoji/EmojiView.java [hunk 11]
```diff
@@ -39,20 +42,35 @@
...
-    themeIconColor = ContextCompat.getColor(context, R.color.emoji_icons);
     final TypedValue value = new TypedValue();
     context.getTheme().resolveAttribute(R.attr.colorAccent, value, true);
     themeAccentColor = value.data;
 
     final ViewPager emojisPager = findViewById(R.id.emojis_pager);
+    final View emojiDivider = findViewById(R.id.emoji_divider);
+    if (dividerColor != 0)
+      emojiDivider.setBackgroundColor(dividerColor);
+    else
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiView.java]: Braces are missing

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java (hunk 9 above)
Comment (RashidianPeyman) [reply emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java]: done

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java (hunk 10 above)
Comment (RashidianPeyman) [reply emoji/src/main/java/com/vanniktech/emoji/EmojiPopup.java]: done

File: emoji/src/main/java/com/vanniktech/emoji/EmojiView.java (hunk 11 above)
Comment (RashidianPeyman) [reply emoji/src/main/java/com/vanniktech/emoji/EmojiView.java]: is it important to have braces?  
You mean that?
 if (dividerColor != 0){
      emojiDivider.setBackgroundColor(dividerColor);
   } else{
      emojiDivider.setBackgroundColor(getResources().getColor(R.color.emoji_divider));
}

**Comment by RashidianPeyman** [comment]
thank you Niklas  its yours and Your hand is open to making the changes

**Comment by RashidianPeyman** [comment]
thank you Niklas 👍👍👍

**Comment by vanniktech** [comment]
There are some findbugs warnings, you want to take a look at it? Otherwise I'll eventually get to it.

**Comment by vanniktech** [comment]
Findbugs is still complaining, I think. Eventually I'll get to it.

**Comment by RashidianPeyman** [comment]
Thank you, Niklas , it's working very well 👍👍

**Pull request merged**